ascanrules: Added Detection for SSTI when a Redirect Occurs

bkollmar opened the pull request:
## Overview
In OWASP Juice Shop, a 302 redirect occurred after the SSTI payload was sent. This caused the POST response to contain information describing the redirect instead of proof that this field was vulnerable to SSTI. By adding one extra line allowing the sendandreceive function to follow the redirect the POST response now contains the changed username that proves there is an SSTI vulnerability and an alert is raised.

## Related Issues
This solves issue #8816

## Checklist
- [ ] Update help
- [ ] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

bkollmar commented:
I have read the CLA Document and I hereby sign the CLA

kingthorin commented:
Please investigate the unit test failures. Your change seems to have impacted some expected behaviour.

You should add test(s) to cover the new behaviour as well.

Also CHANGELOG.md should be updated.

bkollmar committed:
Added if statements to ensure one sendandreceive is called based on status of redirect

created a work around instead of following a redirect to avoid SSRF

fixed formatting

removed getting the Location header

kingthorin commented:
Are you referring to something in the results above from psiinon? You can ignore that.
